### DISCUSSION TITLE
4.1.0rc1

### DISCUSSION THREAD

**Discussion by eschutho** [apache/superset#29705] [General]
Hello Superset Community,

This is a call for the vote to release Apache Superset version 4.1.0.

The release candidate:
https://dist.apache.org/repos/dist/dev/superset/4.1.0rc1/

The Git tag for the release:
https://github.com/apache/superset/tree/4.1.0rc1

The CHANGELOG for the release:
https://github.com/apache/superset/blob/4.1.0rc1/CHANGELOG/4.1.0.md

The instructions for updating to the release:
https://github.com/apache/superset/blob/4.1.0rc1/UPDATING.md

Public keys are available at:
https://www.apache.org/dist/superset/KEYS

<hr /><em>This discussion was created from the release <a href='https://github.com/apache/superset/releases/tag/4.1.0rc1'>4.1.0rc1</a>.</em>

**Comment by padbk** [comment]
I have this bug https://github.com/apache/superset/issues/29708 which is not present in 4.0.2.

**Reply by sadpandajoe** [reply]
Looks like this issue is closed now.

**Reply by sfirke** [reply]
It's a matter of needing to update a config that worked from Superset 2.0.0 - 4.0.2.  I agree it's not a bug, but I wonder if we should link to that issue in the release notes?

**Reply by sadpandajoe** [reply]
@sfirke if you know of the PR that caused this, more that happy to revert it until 5.0 if this is a breaking change. I was looking through the PRs and didn't see one that explicitly changed this functionality but could be due to some other clean up we have in 4.1.

**Reply by sfirke** [reply]
@sadpandajoe I looked through old PRs and I couldn't find one where this changed either 🤷 thanks for looking. I guess if someone figures it out in time we can revert, otherwise I'll just point people to that issue where the fix has been posted

**Reply by wdav** [reply]
It appears that this PR #28158 and commit [2d63722](https://github.com/apache/superset/commit/2d63722150840f93291f3678016f51d94ee72c77#diff-a33df1f92b77f3073377b3b8441027123ea1b7c23edd2c99d14a3754e67cf899L34-L35) removed the imports of `scheduler` and `cache` from `superset/tasks/celery_app.py:L34-35` which causes no task named `reports.scheduler`, etc, to be found if they were not explicitly defined in the config imports.

```diff 
[2d63722] superset/tasks/celery_app.py:L34-35

# Need to import late, as the celery_app will have been setup by "create_app()"
- # pylint: disable=wrong-import-position, unused-import
- from . import cache, scheduler  # isort:skip
```

This can be fixed, as found above, by updating the config to explicitly import it.

But, as there is no context to the removal of this line in the PR, I'm guessing it was not intended?

**Reply by sadpandajoe** [reply]
@mistercrunch I'm guessing this was probably just removed by accident by ruff as they were unused imports. I'll create a PR to address this.

**Reply by sadpandajoe** [reply]
Great find @wdav, I've created the PR: https://github.com/apache/superset/pull/29921 to address this.

**Reply by mistercrunch** [reply]
Oh yes that has to be it. Sorry about that.

**Comment by Habeeb556** [comment]
Hopefully, this bug #28145 can be merged.

**Reply by sadpandajoe** [reply]
Probably not a blocker for 4.1 as this was already happening in older releases.

**Reply by Habeeb556** [reply]
Nope, this hasn't happened in the older releases, especially <= Python 3.8. I was using Superset v2.x, and when I upgraded to Superset v4.x with Python 3.11, this issue occurred.

**Reply by sadpandajoe** [reply]
Looks like Python 3.9 is the supported version right now and we'll bump it to 3.10 next in https://github.com/apache/superset/blob/master/UPDATING.md. Have you tried this when we aren't using 3.11?

**Reply by Habeeb556** [reply]
Sorry, I was using Python 3.8, not 3.9, which is not supported for the upgrade. So, I upgraded to the latest version 3.11.

**Reply by Habeeb556** [reply]
@sadpandajoe, I tried using now a virtual venv with Python 3.10.2 and encountered the same error when checking "Query History":

```plaintext
An error occurred while fetching Query history: 'Template' object has no attribute 'strip'
```

**Reply by ivan-basiuk** [reply]
Hi, will this be fixed in 4.1.0?

**Reply by sadpandajoe** [reply]
Doesn't look like the PR owner has made any updates to their PR. If it gets merged we can try to cherry it in, but I don't see this as a blocker as it's already happening as part of 4.0.2 for some users.

**Reply by Habeeb556** [reply]
++ @toniphan21 #28357

**Comment by iercan** [comment]
Please see below bugs.

#29727 
#29728
#29729

**Reply by eschutho** [reply]
thank you @iercan!

**Reply by sadpandajoe** [reply]
https://github.com/apache/superset/issues/29728 - fixed with https://github.com/apache/superset/pull/29780

**Reply by sadpandajoe** [reply]
Looks like we weren't able to reproduce the issue in https://github.com/apache/superset/issues/29729 so I wouldn't say this is a blocker for 4.1 until we can find something that is reproducible.

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/29789

**Reply by sadpandajoe** [reply]
thanks @michael-s-molina. I'll make sure I cherry this into the next RC

**Reply by sadpandajoe** [reply]
fixed with https://github.com/apache/superset/pull/29799

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/issues/29801

**Reply by michael-s-molina** [reply]
I provided a fix [here](https://github.com/apache/superset/pull/29860).

**Reply by sadpandajoe** [reply]
I think once the PR is merged, we can create RC2 and put it up for another vote.

**Comment by iercan** [comment]
#27416 also still persist with different error. I added comment on issue.

**Reply by sadpandajoe** [reply]
I don't think this is a blocker for 4.1 since it's happening in 3.1.1 and 4.0

**Comment by Habeeb556** [comment]
@eschutho could you push this pre-release `4.1.0rc1` to PyPI for testing?

**Reply by eschutho** [reply]
Since 4.1.0rc1 was downvoted, we'll push one up for rc2.

**Reply by sadpandajoe** [reply]
I'll work on creating a rc2 once https://github.com/apache/superset/pull/29860 is merged and put it up for another round of voting

**Reply by Habeeb556** [reply]
@sadpandajoe please push this pre-release `4.1.0rc2` to PyPI for testing.

**Reply by benvdh-incentro** [reply]
@Habeeb556 It has been released 16 hours ago: https://github.com/apache/superset/releases/tag/4.1.0rc2

The discussion can be found here: https://github.com/apache/superset/discussions/29999

**Reply by Habeeb556** [reply]
Yes I know, but I need it on [PyPI](https://pypi.org/project/apache-superset/#history), so I can `pip install` it.

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/29884

**Reply by sadpandajoe** [reply]
thanks @michael-s-molina for fixing this

**Comment by michael-s-molina** [comment]
https://github.com/apache/superset/pull/29906

**Reply by sadpandajoe** [reply]
will cherry in rc2

**Comment by exolightor** [comment]
In 4.1.0rc1 we are running into this issue which is not present on 4.1.0 nor on master:  [#29931](https://github.com/apache/superset/issues/29931)

**Reply by sadpandajoe** [reply]
@exolightor mind looking again when we cut rc2? 4.1.0rc1 didn't pass and it looks like we are waiting on a couple more fixes before we can cut RC2.
